src/EA.Weee.Web.Tests.Unit/Areas/Admin/Controllers/ProducerSubmissionControllerTests.cs

```diff
@@ -674,6 +674,257 @@ public async Task Removed_Post_NoSubmissions_RedirectsToSubmissions()
+            var result = controller.ReturnProducerRegistration(registrationNumber, year, submission.DirectProducerSubmissionId) as ViewResult;
+
+            breadcrumbService.InternalActivity.Should().Be(InternalUserActivity.DirectRegistrantDetails);
+        }
+
+        [Fact]
+        public async Task ReturnProducerRegistration_Post_SelectValueYes_ReturnsAndRedirectsProducerValues()
+        {
+            // Arrange
+            SetupDefaultControllerData();
+
+            var vm = new ConfirmReturnViewModel
+            {
+                PossibleValues = new[] { string.Empty },
+                SelectedValue = "Yes",
+                Producer = new Core.Admin.ProducerDetailsScheme
+                {
+                    ComplianceYear = 2004,
+                    RegistrationNumber = "regno",
+                    ProducerName = "name",
+                    RegisteredProducerId = Guid.NewGuid()
+                },
+                DirectProducerSubmissionId = Guid.NewGuid()
+            };
+
+            // Act
+            var result = await controller.ReturnProducerRegistration(vm) as RedirectToRouteResult;
+
+            //Assert
+            result.RouteValues["registrationNumber"].Should().Be(vm.Producer.RegistrationNumber);
+            result.RouteValues["producerName"].Should().Be(vm.Producer.ProducerName);
+            result.RouteValues["year"].Should().Be(vm.Producer.ComplianceYear);
+            result.RouteValues["action"].Should().Be(nameof(ProducerSubmissionController.Returned));
+
+            A.CallTo(() => weeeClient.SendAsync(A<string>._,
+            A<ReturnSmallProducerSubmission>.That.Matches(s => s.DirectProducerSubmissionId == vm.DirectProducerSubmissionId)))
+            .MustHaveHappenedOnceExactly();
+        }
+
+        [Fact]
+        public async Task ReturnProducerRegistration_Post_SelectValueNo_ReturnsSubmissionsView()
+        {
+            // Arrange
+            SetupDefaultControllerData();
+
+            var vm = new ConfirmReturnViewModel
+            {
+                PossibleValues = new[] { string.Empty },
+                SelectedValue = "No",
+                Producer = new Core.Admin.ProducerDetailsScheme
+                {
+                    ComplianceYear = 2004,
+                    RegistrationNumber = "regno",
+                    ProducerName = "name",
+                    RegisteredProducerId = Guid.NewGuid()
+                },
+                DirectProducerSubmissionId = Guid.NewGuid()
+            };
+
+            // Act
+            var result = await controller.ReturnProducerRegistration(vm) as RedirectToRouteResult;
+
+            //Assert
+            result.RouteValues["RegistrationNumber"].Should().Be(vm.Producer.RegistrationNumber);
+            result.RouteValues["action"].Should().Be(nameof(ProducerSubmissionController.Submissions));
+        }
+
+        [Fact]
+        public async Task ReturnProducerRegistration_Post_ModelStateInvalid_ReturnsView()
```

Comment:
@will-hope98  when model state isnt valid check that the breadcrumb is reset as it disappears